### ISSUE TITLE
await of GetAsync method never returns and blocks threads

### ISSUE THREAD

**Issue by eduards-v** [EddyCodeIt/My_Listener_Mobile_App#1]
## _Function to Debug_

Function Source: https://github.com/EddyCodeIt/My_Listener_Mobile_App/blob/master/My_Listener/My_Listener/Services/Implementations/NewtonsoftJsonStorageRequests.cs | Line 55

Function call source : https://github.com/EddyCodeIt/My_Listener_Mobile_App/blob/master/My_Listener/My_Listener/MainPage.xaml.cs | line 54

Function: `public async Task<ObservableCollection<TaskTodo>> getToDoList()`
Await operation that hangs: `HttpResponseMessage respon = await objClient.GetAsync(requestUri);`

**Class where broken function is called**

Caller function: `private void MainPage_Loaded(object sender, RoutedEventArgs e)`
Function call: `listOfTasks = requestStorageService.getToDoList().Result;`

## _Notes_

1.  Function call also must await a result from getToDoList(),
but it produce an error : `ObservableCollection<TaskTodo> does not contain a definition for "GetAwaiter" ... `
2. [This article](http://stackoverflow.com/questions/40774931/listmyobject-does-not-contain-a-definition-for-getawaiter) helped to resolve a problem and unblocked the threads.

**Function call looks like this now:** 
```
var listLoadingTask = Task.Run(() => requestStorageService.getToDoList().Result);
listOfTasks = listLoadingTask.Result;
```

Basically what happens is, thread that calls the function spawns new thread to obtain result and freed to continue loading page. In this conditions, debugging shows that GetAsync request does not hang anymore and function successfully returns. 

3. A new problem arises. Observable collection, that is meant to be loaded with data from a Task here `listOfTasks = listLoadingTask.Result;` is not displayed in UI. 
I presume  UI is not notified that collection is finally loaded with data when task returns. 
How to notify collection about incoming changes?

**Comment by eduards-v** [comment]
Change to MVVM architecture and x:Bind feature solved an issue.
